client/src/charts/nivo/nivo_shared.js

```diff
@@ -0,0 +1,195 @@
+    <td className="nivo-tooltip__content" dangerouslySetInnerHTML = {{__html: formats.percentage1(Math.abs(tooltip_item.value)/total)}}/>
+  </Fragment>
+);
+
+const default_tooltip = (tooltip_items, formatter, total) => ( // total indicates percent value tooltip being used
+  <div style={{color: window.infobase_color_constants.textColor}}>
+    <table className="nivo-tooltip">
+      <tbody>
+        { tooltip_items.map(
+          tooltip_item => ( 
+            <tr key = {tooltip_item.id}>
+              <td className="nivo-tooltip__content">
+                <div style={{height: '12px', width: '12px', backgroundColor: tooltip_item.color}} />
+              </td>
+              <MediaQuery minDeviceWidth={breakpoints.minSmallDevice}>
+                {total ? percent_tooltip_content(tooltip_item, formatter, total) : tooltip_content(tooltip_item, formatter) }
+              </MediaQuery>
+              <MediaQuery maxDeviceWidth={breakpoints.maxSmallDevice}>
+                {total ? smalldevice_percent_tooltip_content(tooltip_item, formatter, total) : smalldevice_tooltip_content(tooltip_item, formatter)}
+              </MediaQuery>
+            </tr>
+          )
+        )}
+      </tbody>
+    </table>
+  </div>
+);
+
+class InteractiveGraph extends React.Component{
+  constructor(props){
+    super(props);
+    this.state = {
+      show_table: false,
+    };
+  }
+
+  render(){      
+    const {
+      show_table,
+    } = this.state;
+
+    const {
+      graph,
+      table,
+      table_name,
+      other_buttons,
+    } = this.props;
+
+    return (
+      <Fragment>
+        <div> {/* Don't get rid of this, need it for proper functioning of the child selectors on the buttons */}
+          { table && 
+            <button
+              className={classNames("btn-ib-primary","btn-ib-array")}
+              style={{
+                zIndex: 999,
+              }}
+              onClick={ () => this.setState({ show_table: !show_table }) }
+            >
+              <IconTable
+                title={graph_text_maker("show_table")}
+                color={window.infobase_color_constants.secondaryColor}
+                alternate_color={window.infobase_color_constants.backgroundColor}
+              />
+            </button>
+          }
+          { _.map(other_buttons, (button,i) => <Fragment key={i}>{button}</Fragment> ) }
+        </div>
+        { graph }
+        <StatelessModal
+          show={ show_table }
+          title={ table_name || graph_text_maker("default_table_name") }
+          body={ table }
+          on_close_callback={() => this.setState({show_table: false})}
+          additional_dialog_class = { (() => {
+            if (window.innerWidth < breakpoints.minMediumDevice) { return 'modal-sm'; }
+            if (window.innerWidth > breakpoints.maxLargeDevice ) { return 'modal-xl'; }
+            return 'modal-lg';
+          })() }
```

Comment:
This isn't going to respond to window resizes. Probably fine since since it will be re-called every time the modal's open/closed, but it would be preferable if it was automatic (I could see someone clicking the button on their phone, then deciding to rotate to try and see more of the table, which might not work great as-is).

You don't have to work that out now, but if you don't then please open an issue about it when this PR merges.